src/btree/bt_discard.c

```diff
@@ -245,20 +245,31 @@ __free_page_modify(WT_SESSION_IMPL *session, WT_PAGE *page)
 void
 __wt_ref_addr_free(WT_SESSION_IMPL *session, WT_REF *ref)
 {
+    WT_PAGE *home;
     void *ref_addr;
 
+    home = ref->home;
+
     /*
-     * The page being discarded may be the child of a page being split, where the WT_REF.addr field
-     * is being instantiated (as it can no longer reference the on-disk image). Loop until we read
-     * and clear the address without a race, then free the read address as necessary.
+     * In order to free the WT_REF.addr field we need to read and clear the address without a race.
+     * The WT_REF may be a child of a page being split, in which case the addr field could be
+     * instantiated concurrently which changes the addr field. Once we swap in NULL we effectively
+     * own the addr. Then provided the addr is off page we can free the memory.
+     *
+     * However as we could be the child of a page being split the ref->home pointer which tells us
+     * whether the addr is on or off page could change concurrently. To avoid this we save the home
+     * pointer before we do the compare and swap.
      */
     do {
         WT_ORDERED_READ(ref_addr, ref->addr);
         if (ref_addr == NULL)
             return;
     } while (!__wt_atomic_cas_ptr(&ref->addr, ref_addr, NULL));
 
-    if (ref->home == NULL || __wt_off_page(ref->home, ref_addr)) {
+    /* Encourage races. */
```

Comment:
I would add "Encourage races in test format." just to make it super obvious off the bat.

Reply:
Timing stress configurations are enabled via the WT open configuration and are not exclusive to test format.

Reply:
I see, is it used for more than just testing?

Reply:
It shouldn't be used outside of testing but we don't stop it from being used.